src/js/player.js

```diff
@@ -1255,6 +1266,24 @@ class Player extends Component {
     return;
   }
 
+  middlewareIterator(method) {
+    return (value, mw) => {
+      if (mw[method]) {
+        return mw[method](value);
+      }
+
+      return value;
+    };
+  }
+
+  middlewareSet(method, arg) {
+    return this.middleware_.reduce(this.middlewareIterator(method), arg);
+  }
+
+  middlewareGet(method) {
+    return this.middleware_.reduceRight(this.middlewareIterator(method), undefined);
+  }
```

Comment:
The methods should be private. The reason why they aren't in `middleware.js` and why `setSource` is special is because `setSource` traverses through our middleware tree and sets up the specific middleware to be used for the player at this time. Though, I guess it can be moved into `middleware.js` and then `this.middleware_` could be passed in.